src/pages/RFOX/components/RewardsAndClaims/Rewards.tsx

```diff
@@ -1,32 +1,126 @@
 import { Box, CardBody } from '@chakra-ui/react'
-import { thorchainAssetId } from '@shapeshiftoss/caip'
-import { TxStatus } from '@shapeshiftoss/unchained-client'
+import type { AccountId } from '@shapeshiftoss/caip'
+import { fromAccountId, thorchainAssetId, thorchainChainId, toAccountId } from '@shapeshiftoss/caip'
+import { DAO_TREASURY_THORCHAIN } from '@shapeshiftoss/utils'
+import { getConfig } from 'config'
+import { uniq } from 'lodash'
 import { useMemo } from 'react'
+import { Text } from 'components/Text'
 import { TransactionsGroupByDate } from 'components/TransactionHistory/TransactionsGroupByDate'
-import { selectTxIdsByFilter } from 'state/slices/selectors'
+import { isSome } from 'lib/utils'
+import { selectRuneAddress } from 'pages/RFOX/helpers'
+import { useStakingInfoHistoryQuery } from 'pages/RFOX/hooks/useStakingInfoHistoryQuery'
+import { useStakingInfoQuery } from 'pages/RFOX/hooks/useStakingInfoQuery'
+import { selectReceivedTxsForAccountIdsByFilter } from 'state/slices/selectors'
 import { useAppSelector } from 'state/store'
 
+type RewardsContentProps = {
+  stakingAssetAccountId: AccountId
+}
+
 type RewardsProps = {
+  stakingAssetAccountId: AccountId
   headerComponent: JSX.Element
 }
 
-export const Rewards = ({ headerComponent }: RewardsProps) => {
-  const filter = useMemo(
-    () => ({ txStatus: TxStatus.Confirmed, assetIdFilter: thorchainAssetId }),
-    [],
+const RewardsContent = ({ stakingAssetAccountId }: RewardsContentProps) => {
+  const stakingAssetAccountAddress = useMemo(
+    () => fromAccountId(stakingAssetAccountId).account,
+    [stakingAssetAccountId],
   )
-  const txIds = useAppSelector(state => selectTxIdsByFilter(state, filter))
+
+  const {
+    data: historicalRuneAddresses,
+    isLoading: isStakingInfoHistoryLoading,
+    isFetching: isStakingInfoHistoryFetching,
+  } = useStakingInfoHistoryQuery({ stakingAssetAccountAddress, select: selectRuneAddress })
+
+  const {
+    data: currentRuneAddress,
+    isLoading: isCurrentStakingInfoLoading,
+    isFetching: isCurrentStakingInfoFetching,
+  } = useStakingInfoQuery({ stakingAssetAccountAddress, select: selectRuneAddress })
+
+  const uniqueHistoricalRuneAddresses = useMemo(() => {
```

Comment:
This query internally uses `useQueries` where `select` is defined on a singular query output, so we cannot select on the overall output to reduce all results down to a unique set and instead must do it externally here

Reply:
Alternatively we could force that API by adding `select` to `combine`, but this would be diverging from the react-query approach to this type of query...  but on the other hand it makes more sense to be consistent

I'll leave as-is for now and if we need it to change we can do it then